pkg/registry/thirdpartyresourcedata/codec.go

```diff
@@ -207,40 +208,49 @@ const template = `{
   "items": [ %s ]
 }`
 
-func encodeToJSON(obj *experimental.ThirdPartyResourceData) ([]byte, error) {
+func encodeToJSON(obj *experimental.ThirdPartyResourceData, stream io.Writer) error {
 	var objOut interface{}
 	if err := json.Unmarshal(obj.Data, &objOut); err != nil {
-		return nil, err
+		return err
 	}
 	objMap, ok := objOut.(map[string]interface{})
 	if !ok {
-		return nil, fmt.Errorf("unexpected type: %v", objOut)
+		return fmt.Errorf("unexpected type: %v", objOut)
 	}
 	objMap["metadata"] = obj.ObjectMeta
-	return json.Marshal(objMap)
+	encoder := json.NewEncoder(stream)
+	return encoder.Encode(objMap)
+}
+
+func (t *thirdPartyResourceDataCodec) Encode(obj runtime.Object) ([]byte, error) {
+	buff := &bytes.Buffer{}
```

Comment:
Actually it is:
```sh
_output/local/go/src/k8s.io/kubernetes/pkg/registry/thirdpartyresourcedata/codec.go:242: cannot use buff (type bytes.Buffer) as type io.Writer in argument to encodeToJSON:
	bytes.Buffer does not implement io.Writer (Write method has pointer receiver)
```

I could take the pointer below where I pass it into the Encoder, but that seems like 6 of one 1/2 dozen of the other.